_docs/aop.md

```diff
@@ -112,36 +114,43 @@ The classloader configuration might look like the following.
 
-You might additionally have a look at [this](https://en.wikipedia.org/wiki/Aspect-oriented_programming) and [this](http://eclipse.org/aspectj/doc/released/progguide/starting-aspectj.html) link for further explanations. 
+There is an analogy useful for understanding AOP and it involves a prism. With a prism, you can dissect a ray of light into its individual aspects (different colors), and you can use another prism to bring those colors back to the ray of light. The same effect is what AOP offers, as a programming paradigm. You dissect your problem into its singular aspects and can weave them back together to form the solution to the whole problem.
 
 ### Join-points
-It is already clear that AOP is about the insertion (one speaks of weaving) of a code (logic of our `cross-cutting concerns`) into defined points within a program's flow. It is also obvious that we have to describe this point and which concern we want to insert.
+It is hopefully clear to you now that AOP is about the insertion (one speaks of weaving) of code (logic of our `cross-cutting concerns`) into defined points within a program's flow. It should also be obvious that we have to describe these points and which concerns we want to insert.
 
-`Join points` are used to describe a certain event within the actual flow of a running program at which weaving should take place. 
+`Join points` are used to describe a certain event within the actual flow of a running program, at which weaving should take place.
 
 > So, rather than describing a place within the source code, join points describe something that is *happening* at a specific place in your application.
- 
- We currently support two join points, one standalone and the other one as addition for other join points.
- 
+
+ We currently support two join point types.
+
 | Keyword      | Type       | Description                                                     |
 | -------------| -----------| ----------------------------------------------------------------|
 | `call`       | standalone | The call of a method being static or dynamic.                   |
-| `execute`    | addition   | An addition to other join points. Describes that a certain join point has to be in the context of an execution stack. E.g. a call to `methodB` might only be interesting if it was made from within `methodA` |
+| `execute`    | addition   | An addition to other join points. Describes that a certain join point has to be in the context of an execution stack. E.g. a call to `methodB` will only be carried out, if it was made from within `methodA` |
```

Comment:
E.g. when a call to methodB is made from within methodA, then the execute join point is a match and the woven advice will be carried out.